src/octoprint/settings/__init__.py

```diff
@@ -1769,8 +1774,9 @@ def remove(self, path, config=None, error_on_path=False, defaults=None):
         chain = self._map.with_config_defaults(config=config, defaults=defaults)
 
         try:
-            chain.del_by_path(path)
```

Comment:
I'm not familiar with this logic or how it works at all, but would `self._map.with_config_defaults` also need to be in the lock? Or would that be safe to execute from multiple threads?

Reply:
That's a very good question that I have currently no answer on and that needs some digging. Probably a thing for after my vacation.

Reply:
Have given that some more thought and looking at now, and I don't think we need that. Building the new chain to work on is pure recombining of a view on existing data structures, without any modification taking place as far as I can see. So that should be inherently thread safe already and thus doesn't need to be pulled into the lock. The goal here is to make sure we don't write an inconsistent state to disk due to concurrent modification from two threads, and that should be ensured by the existing changes.